tests/ibc-hooks/ibc_middleware_test.go

```diff
@@ -618,14 +639,178 @@ func (suite *HooksTestSuite) SetupCrosschainSwaps(chainName Chain) (sdk.AccAddre
+	suite.Require().NoError(err)
+}
+
+// modifyChainChannelLinks modifies the chain channel links in the crosschain registry utilizing set, remove, and change operations
+func (suite *HooksTestSuite) modifyChainChannelLinks(registryAddr sdk.AccAddress, chainName Chain) {
+	chain := suite.GetChain(chainName)
+	ctx := chain.GetContext()
+	owner := chain.SenderAccount.GetAddress()
+	osmosisApp := chain.GetOsmosisApp()
+	contractKeeper := wasmkeeper.NewDefaultPermissionKeeper(osmosisApp.WasmKeeper)
+
+	msg := `{
+		"modify_chain_channel_links": {
+		  "operations": [
+			{"operation": "remove","source_chain": "chainB","destination_chain": "chainC","channel_id": "channel-1"},
+			{"operation": "set","source_chain": "ChainB","destination_chain": "ChainC","channel_id": "channel-1"},
+			{"operation": "remove","source_chain": "chainC","destination_chain": "chainB","channel_id": "channel-0"},
+			{"operation": "set","source_chain": "ChainC","destination_chain": "ChainB","channel_id": "channel-0"},
+			{"operation": "change","source_chain": "chainB","destination_chain": "osmosis","new_source_chain": "ChainB"},
+			{"operation": "change","source_chain": "osmosis","destination_chain": "chainB","new_destination_chain": "ChainB"}
+		  ]
+		}
+	  }
+	  `
+	_, err := contractKeeper.Execute(ctx, registryAddr, owner, []byte(msg), sdk.NewCoins())
+	suite.Require().NoError(err)
+}
+
+// modifyChainChannelLinks modifies the chain channel links in the crosschain registry utilizing set, remove, and change operations
+func (suite *HooksTestSuite) setContractAlias(registryAddr sdk.AccAddress, contractAlias string, chainName Chain) {
+	chain := suite.GetChain(chainName)
+	ctx := chain.GetContext()
+	owner := chain.SenderAccount.GetAddress()
+	osmosisApp := chain.GetOsmosisApp()
+	contractKeeper := wasmkeeper.NewDefaultPermissionKeeper(osmosisApp.WasmKeeper)
+
+	msg := fmt.Sprintf(`{
+		"modify_contract_alias": {
+		  "operations": [
+			{"operation": "set", "alias": "%s", "address": "%s"}
+		  ]
+		}
+	  }
+	  `, contractAlias, registryAddr)
+	_, err := contractKeeper.Execute(ctx, registryAddr, owner, []byte(msg), sdk.NewCoins())
+	suite.Require().NoError(err)
+}
+
+func (suite *HooksTestSuite) TestCrosschainRegistry() {
+	// Instantiate contract and set up three chains with funds sent between each
+	registryAddr, token0CB, token0CBA, CBAPath := suite.SetupCrosschainRegistry(ChainA)
+
+	// Set the registry address to an alias
+	contractAlias := "osmosis_registry_contract"
+	suite.setContractAlias(registryAddr, contractAlias, ChainA)
+
+	// Retrieve the registry address from the alias
+	contractAddressFromAliasQuery := fmt.Sprintf(`{"get_address_from_alias": {"contract_alias": "%s"}}`, contractAlias)
+	contractAddressFromAliasQueryResponse := suite.chainA.QueryContract(&suite.Suite, registryAddr, []byte(contractAddressFromAliasQuery))
+	suite.Require().Equal(fmt.Sprintf("\"%s\"", registryAddr.String()), contractAddressFromAliasQueryResponse)
```

Comment:
Note to reviewer: I have no idea why the address is returning with an extra set of quotes and I have no idea how to get rid of them in a non hacky way

Reply:
I'll review in more detail in a bit, but I think this is because we are json-serializing the string before returning it

Reply:
Right, so this is what we expect. Cosmwasm queries results as json encoded. We may want to create a struct so that they're return inside an object `{"contract_address": "..."}`

Reply:
Got it, fixed here! https://github.com/osmosis-labs/osmosis/pull/4320/commits/64a70672fd9576013e35d625dff4615b3a799640